sdcm/utils/azure_utils.py

```diff
@@ -0,0 +1,216 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2021 ScyllaDB
+
+from __future__ import annotations
+
+import logging
+from typing import NamedTuple, TYPE_CHECKING
+from functools import cached_property
+from itertools import chain
+
+from azure.identity import ClientSecretCredential
+from azure.mgmt.compute import ComputeManagementClient
+from azure.mgmt.compute.models import VirtualMachine
+from azure.mgmt.network import NetworkManagementClient
+from azure.mgmt.resource import ResourceManagementClient
+from azure.mgmt.subscription import SubscriptionClient
+from azure.mgmt.resourcegraph import ResourceGraphClient
+from azure.mgmt.resourcegraph.models import QueryRequestOptions, QueryRequest
+
+from sdcm.keystore import KeyStore
+from sdcm.utils.metaclasses import Singleton
+
+if TYPE_CHECKING:
+    # pylint: disable=ungrouped-imports
+    from typing import Optional, Callable, Iterator
+
+    from azure.core.credentials import TokenCredential
+    from azure.mgmt.resource.resources.models import Resource
+
+
+API_VERSIONS = {
+    "Microsoft.Compute/disks": "2020-12-01",
+    "Microsoft.Compute/galleries": "2021-07-01",
+    "Microsoft.Compute/galleries/images": "2021-07-01",
+    "Microsoft.Compute/galleries/images/versions": "2021-07-01",
+    "Microsoft.Compute/virtualMachines": "2021-07-01",
+    "Microsoft.Network/networkInterfaces": "2021-02-01",
+    "Microsoft.Network/networkSecurityGroups": "2021-02-01",
+    "Microsoft.Network/publicIPAddresses": "2021-02-01",
+    "Microsoft.Network/virtualNetworks": "2021-02-01",
+    'Microsoft.Network/virtualNetworks/subnets': "2021-02-01",
+    "Microsoft.Resources/resourceGroups": "2021-04-01",
+}
+
+LOGGER = logging.getLogger(__name__)
+
+
+logging.getLogger("azure").setLevel(logging.ERROR)
```

Comment:
Check how we can define the level in `sdcm/utils/log.py`

Reply:
Unfortunately, we need to make too many changes to use `conifgure_logging()` in `sct.py`